services/assetService.js

```diff
@@ -113,40 +107,34 @@ AssetService.prototype.findByPath = function(path) {
 }
 
 AssetService.prototype.save = function(data, user) {
-	var that = this;
+	var that = this
 
 	return this.findByPath(data.path)
 	.then(function(asset) {
-		if (!asset)
-			asset = new that._model(data);
-
-		asset._creator = user.id;
-		asset.updatedAt = Date.now();
-
-		if (data.tags)
-			asset.tags = data.tags
-
-		if (data.previewUrlSmall)
-			asset.previewUrlSmall = data.previewUrlSmall
+		if (!asset) {
+			asset = new that._model(data)
+			asset._creator = user.id
+		}
 
-		if (data.previewUrlLarge)
-			asset.previewUrlLarge = data.previewUrlLarge
+		// update model with stuff from data
+		_.assign(asset, data)
```

Comment:
so passing data with existing but empty keys will now overwrite the asset? double-checking this is desired.